var/spack/repos/builtin/packages/espresso/package.py

```diff
@@ -91,38 +94,62 @@ def install(self, spec, prefix):
             options.append('--enable-openmp')
 
         if '+scalapack' in spec:
-            options.append('--with-scalapack=yes')
+            scalapack_option = 'intel' if '%intel' in spec else 'yes'
+            options.append('--with-scalapack={0}'.format(scalapack_option))
 
         if '+elpa' in spec:
-            if spec.satisfies('@:5.4.0'):
-                options.append('--with-elpa=yes')
-            else:
-                options.extend([
-                    '--with-elpa-include={0}'.format(
-                        join_path(spec['elpa'].prefix,
-                                  'include',
-                                  'elpa-{0}'.format(str(spec['elpa'].version)),
-                                  'modules')
-                    ),
-                    '--with-elpa-lib={0}'.format(spec['elpa'].libs.joined())
-                ])
+
+            # Spec for elpa
+            elpa = spec['elpa']
+
+            # Find where the Fortran module resides
+            elpa_module = find(elpa.prefix, 'elpa.mod')
+
+            # Compute the include directory from there: versions
+            # of espresso prior to 6.1 requires -I in from of the directory
+            elpa_include = '' if '@6.1:' in spec else '-I'
+            elpa_include += os.path.dirname(elpa_module[0])
+            options.append('--with-elpa-include={0}'.format(elpa_include))
+
+            # Search for the libraries to link
+            library = 'libelpa_openmp' if '+openmp' in spec else 'libelpa'
+            elpa_libraries = find_libraries(
+                library, elpa.prefix, recurse=True
+            )
+            options.append(
+                '--with-elpa-lib={0}'.format(elpa_libraries[0])
+            )
+
+        if '+hdf5' in spec:
+            options.append('--with-hdf5={0}'.format(spec['hdf5'].prefix))
+
+        # Compiler related configure options
+        if '%intel' in self.spec:
```

Comment:
that look hacky:
1. i may still be using `openmpi` with `intel` compiler. 
2. it should be perfectly fine to ask `mpi` for the path to f90 instead of hard-coding `mpiifort` here.